vt/client.go

```diff
@@ -222,6 +223,30 @@ func (cli *Client) DownloadFile(hash string, w io.Writer) (int64, error) {
 	return io.Copy(w, resp.Body)
 }
 
+// GetSignedFileDownloadURL returns a signed URL for downloading
+func (cli *Client) GetSignedFileDownloadURL(hash string) (string, error) {
+	u := URL("files/%s/download", hash)
+
+	cli.httpClient.CheckRedirect = func(req *http.Request, via []*http.Request) error {
```

Comment:
The same client could be used concurrently from different goroutines, changing CheckRedirect may have side effects on those goroutines. Even if you restore it to the previous value there is a time interval where CheckRedirect changes for everyone using the client.

What you are looking for is the [/files/{id}/download_url](https://developers.virustotal.com/v3.0/reference?showHidden=956bd#files-download-url) endpoint.